commit 61d136ff9eec815d88cc8d4c3de99a6c850bad14
Author: Gerald Pape <ubergesundheit@users.noreply.github.com>
Date:   Tue Nov 21 15:32:11 2017 +0100

    enable sht3x module in kernel, add sht3x definition to i2c-sensor dts definition (#2282)
    
    Enable sht3x module in kernel configuration, and add to i2c-sensor overlay.

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 68496003d906..a162494d5344 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -746,7 +746,7 @@ Info:   Adds support for a number of I2C barometric pressure and temperature
         sensors on i2c_arm
 Load:   dtoverlay=i2c-sensor,<param>=<val>
 Params: addr                    Set the address for the BME280, BMP280, TMP102,
-                                HDC100X or LM75
+                                HDC100X, LM75 or SHT3x
 
         bme280                  Select the Bosch Sensortronic BME280
                                 Valid addresses 0x76-0x77, default 0x76
@@ -780,6 +780,10 @@ Params: addr                    Set the address for the BME280, BMP280, TMP102,
         veml6070                Select the Vishay VEML6070 ultraviolet light
                                 sensor
 
+        sht3x                   Select the Sensiron SHT3x temperature and
+                                humidity sensor. Valid addresses 0x44-0x45,
+                                default 0x44
+
 
 Name:   i2c0-bcm2708
 Info:   Enable the i2c_bcm2708 driver for the i2c0 bus. Not all pin combinations
diff --git a/arch/arm/boot/dts/overlays/i2c-sensor-overlay.dts b/arch/arm/boot/dts/overlays/i2c-sensor-overlay.dts
index 28e6fa3b659d..0b9bc8ff9dd2 100644
--- a/arch/arm/boot/dts/overlays/i2c-sensor-overlay.dts
+++ b/arch/arm/boot/dts/overlays/i2c-sensor-overlay.dts
@@ -171,9 +171,24 @@
 		};
 	};
 
+	fragment@11 {
+		target = <&i2c_arm>;
+		__dormant__ {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			status = "okay";
+
+			sht3x: sht3x@44 {
+				compatible = "sht3x";
+				reg = <0x44>;
+				status = "okay";
+			};
+		};
+	};
+
 	__overrides__ {
 		addr =  <&bme280>,"reg:0", <&bmp280>,"reg:0", <&tmp102>,"reg:0",
-			<&lm75>,"reg:0",<&hdc100x>,"reg:0";
+			<&lm75>,"reg:0", <&hdc100x>,"reg:0", <&sht3x>,"reg:0";
 		bme280 = <0>,"+0";
 		bmp085 = <0>,"+1";
 		bmp180 = <0>,"+2";
@@ -186,5 +201,6 @@
 		hdc100x = <0>,"+8";
 		tsl4531 = <0>,"+9";
 		veml6070 = <0>,"+10";
+		sht3x = <0>,"+11";
 	};
 };
diff --git a/arch/arm/configs/bcm2709_defconfig b/arch/arm/configs/bcm2709_defconfig
index fea5aef1d155..dc8b77f0637d 100644
--- a/arch/arm/configs/bcm2709_defconfig
+++ b/arch/arm/configs/bcm2709_defconfig
@@ -662,6 +662,7 @@ CONFIG_HWMON=m
 CONFIG_SENSORS_JC42=m
 CONFIG_SENSORS_LM75=m
 CONFIG_SENSORS_SHT21=m
+CONFIG_SENSORS_SHT3x=m
 CONFIG_SENSORS_SHTC1=m
 CONFIG_SENSORS_ADS1015=m
 CONFIG_SENSORS_INA2XX=m
diff --git a/arch/arm/configs/bcmrpi_defconfig b/arch/arm/configs/bcmrpi_defconfig
index 4888aefa63d8..2e5a841c54ea 100644
--- a/arch/arm/configs/bcmrpi_defconfig
+++ b/arch/arm/configs/bcmrpi_defconfig
@@ -656,6 +656,7 @@ CONFIG_HWMON=m
 CONFIG_SENSORS_JC42=m
 CONFIG_SENSORS_LM75=m
 CONFIG_SENSORS_SHT21=m
+CONFIG_SENSORS_SHT3x=m
 CONFIG_SENSORS_SHTC1=m
 CONFIG_SENSORS_ADS1015=m
 CONFIG_SENSORS_INA2XX=m
